internal/datastore/common/changes.go

```diff
@@ -85,18 +189,29 @@ func (ch Changes[R, K]) AsRevisionChanges(lessThanFunc func(lhs, rhs K) bool) []
 		revisionChangeRecord := ch.records[k]
 		changes[i].Revision = revisionChangeRecord.rev
 		for _, tpl := range revisionChangeRecord.tupleTouches {
-			changes[i].Changes = append(changes[i].Changes, &core.RelationTupleUpdate{
+			changes[i].RelationshipChanges = append(changes[i].RelationshipChanges, &core.RelationTupleUpdate{
 				Operation: core.RelationTupleUpdate_TOUCH,
 				Tuple:     tpl,
 			})
 		}
 		for _, tpl := range revisionChangeRecord.tupleDeletes {
-			changes[i].Changes = append(changes[i].Changes, &core.RelationTupleUpdate{
+			changes[i].RelationshipChanges = append(changes[i].RelationshipChanges, &core.RelationTupleUpdate{
 				Operation: core.RelationTupleUpdate_DELETE,
 				Tuple:     tpl,
 			})
 		}
+		changes[i].ChangedDefinitions = maps.Values(revisionChangeRecord.definitionsChanged)
+		changes[i].DeletedNamespaces = maps.Keys(revisionChangeRecord.namespacesDeleted)
+		changes[i].DeletedCaveats = maps.Keys(revisionChangeRecord.caveatsDeleted)
```

Comment:
you could cache both `maps.Keys` calls to namespaces and caveats, and then append them to `ChangedDefinitions`. It would spare the need of keeping `revisionChangeRecord.definitionsChanged` at all.

Reply:
? Not following. Changed definitions do not include those deleted

Reply:
ah doh, my bad